src/plugins/integrations/HttpIntegration.ts

```diff
@@ -50,6 +63,16 @@ class HttpIntegration implements Integration {
           disableActiveStart: true,
         });
 
+        if (!(Utils.getConfiguration(envVariableKeys.DISABLE_SPAN_CONTEXT_INJECTION) === 'true') ) {
+          const headers = options.headers ? options.headers : {};
+          tracer.inject(span.spanContext, opentracing.FORMAT_TEXT_MAP, headers);
+          options.headers = headers;
+        }
+
+        if (!HttpIntegration.isValidUrl(host)) {
+          return request.apply(this, [options, callback]);
```

Comment:
I think, in case of invalid url that we are not interested in, headers should not be passed in the `options`

Reply:
Done, if the url is invalid we do not create span and inject any headers.